core/src/main/java/org/lflang/generator/docker/CppDockerGenerator.java

```diff
@@ -0,0 +1,93 @@
+package org.lflang.generator.docker;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+import org.lflang.generator.LFGeneratorContext;
+import org.lflang.generator.cpp.CppPlatformGenerator;
+import org.lflang.generator.cpp.CppStandaloneGenerator;
+import org.lflang.target.property.BuildTypeProperty;
+
+/**
+ * Generates the docker file related code for the C++ target.
+ *
+ * @author Marten Lohstroh
+ */
+public class CppDockerGenerator extends DockerGenerator {
+
+  private CppPlatformGenerator platformGenerator;
+
+  /** Construct a new Docker generator. */
+  public CppDockerGenerator(LFGeneratorContext context, CppPlatformGenerator platformGenerator) {
+    super(context);
+    this.platformGenerator = platformGenerator;
+  }
+
+  @Override
+  protected String generateCopyForSources() {
+    return "COPY src-gen src-gen";
+  }
+
+  public static final String DEFAULT_BASE_IMAGE = "alpine:latest";
+
+  @Override
+  public String defaultImage() {
+    return DEFAULT_BASE_IMAGE;
+  }
+
+  @Override
+  protected String generateRunForInstallingDeps() {
+    if (builderBase().equals(defaultImage())) {
+      return "RUN set -ex && apk add --no-cache g++ musl-dev cmake make && apk add --no-cache"
+          + " --update --repository=https://dl-cdn.alpinelinux.org/alpine/v3.16/main/"
+          + " libexecinfo-dev";
+    } else {
+      return "# (Skipping installation of build dependencies; custom base image.)";
+    }
+  }
+
+  @Override
+  public List<String> defaultEntryPoint() {
+    return List.of("./bin/" + context.getFileConfig().name);
+  }
+
+  @Override
+  protected List<String> defaultBuildCommands() {
+    var mkdirCommand = List.of("mkdir", "-p", "build", "&&", "mkdir", "-p", "bin");
+    var cmakeCommand = new ArrayList<String>();
+    cmakeCommand.add("cmake");
+    cmakeCommand.addAll(platformGenerator.getCmakeArgs());
+    cmakeCommand.addAll(
+        List.of(
+            "-DCMAKE_INSTALL_BINDIR=bin",
+            "-DCMAKE_INSTALL_PREFIX=.",
+            "-DREACTOR_CPP_LINK_EXECINFO=ON",
+            "-S",
+            "src-gen",
+            "-B",
+            "build"));
+    var makeCommand =
```

Comment:
Duplicating the complete commands is quite unfortunate, as the commands used here and in the normal build will get out of sync. Also, this does not for the ROS2 support of the C++ target. I think a better approach would be to extend the platform generator interface, adding a method for querying the commands that are used there. Alternatively, we could also specialize the docker generator for the standalone and the ros2 platform and have a `getDockerGenerator` method as part of the platform generator interface.

Reply:
OK, this sounds like it should work.

Reply:
I ran out of time before I got the build to work on ROS. What needs to be done there is that there needs to be a special Configurator for ROS 2 support + Docker support in the test framework, and the ROS 2 bash script needs to somehow be copied into the Docker image. I could do that tomorrow if we think that it is important.

Reply:
Great effort, @petervdonovan. This brings us really close.